snapcraft/storeapi/errors.py

```diff
@@ -111,28 +111,63 @@ def __init__(self, exception):
         super().__init__(message=message)
 
 
-class SnapNotFoundError(StoreError):
+class SnapNotFoundError(SnapcraftException):
+    def __init__(
+        self,
+        *,
+        snap_name: str = "",
+        snap_id: Optional[str] = None,
+        channel: Optional[str] = None,
+        arch: Optional[str] = None,
+    ):
+        super().__init__()
 
-    __FMT_ARCH_CHANNEL = (
-        "Snap {name!r} for {arch!r} cannot be found in the {channel!r} channel."
-    )
-    __FMT_CHANNEL = "Snap {name!r} was not found in the {channel!r} channel."
-    __FMT_SERIES_ARCH = "Snap {name!r} for {arch!r} was not found in {series!r} series."
-    __FMT_SERIES = "Snap {name!r} was not found in {series!r} series."
+        # Defaulting snap_name to "" to support the one case we have that
+        # makes use of snap_id.
+        if snap_name == "" and snap_id is None:
```

Comment:
Just a thought:  you could change the caller w/ snap_id to pass in something like snap_name=f"id {snap_id!r}" and drop the special handling.

Reply:
Not sure I am ok with making the internal representation be a reflection of what we want for presentation.

Reply:
Fair enough